src/shared/highlighted-field/use-highlighted-field.js

```diff
@@ -49,10 +49,17 @@ export default function useHighlightedField() {
 
     useEffect(() => {
         const { de, coc } = item
-        setHighlightedFieldId(
-            gatherHighlightedFieldData({ de, coc, dataValueSet })
-        )
-    }, [item, dataValueSet])
+        // only update if it's different from currentItem
+        // prevents unnecessary re-render on first-render as well
+        if (
+            de.id !== currentItem?.dataElement ||
+            coc.id !== currentItem?.categoryOptionCombo
+        ) {
+            setHighlightedFieldId(
+                gatherHighlightedFieldData({ de, coc, dataValueSet })
+            )
+        }
+    }, [currentItem, item, dataValueSet])
```

Comment:
I noticed that this was causing an additional render, and was the "cause" of the duplicated request. However, I wouldn't think of this as a proper solution, since we shouldn't have to worry about excess request due to excessive renders.